src/libraries/System.Transactions.Local/src/System/Transactions/DtcProxyShim/DTCInterfaces/ITransactionOutcomeEvents.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Transactions.Oletx;
+
+namespace System.Transactions.DtcProxyShim.DTCInterfaces;
+
+/// <summary>
+/// This interface is used by application programs that require asynchronous notification about transaction outcomes.
+/// The application program implements the methods in this interface and registers the interface with the connection point mechanism.
+/// DTC calls the appropriate method on this interface to inform the application about the outcome of a transaction.
+/// </summary>
+/// <remarks>
+/// See https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ms686465(v=vs.85).
+/// </remarks>
+[ComImport, Guid("3A6AD9E2-23B9-11cf-AD60-00AA00A74CCD"), InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+internal interface ITransactionOutcomeEvents
+{
+    /// <summary>
+    /// This event is raised when the transaction committed.
+    /// </summary>
+    /// <param name="fRetaining">Indicates whether retaining Commit was specified. Will be false.</param>
+    /// <param name="pNewUOW">Always null.</param>
+    /// <param name="hresult">Always S_OK.</param>
+    void Committed(bool fRetaining, Guid pNewUOW /* always null? */, uint hresult);
+
+    /// <summary>
+    /// This event is raised when the transaction aborted, either as a result of a call to Abort or an unsuccessful call to Commit*.*.
+    /// </summary>
+    /// <param name="pboidReason">A BOID indicating why the transaction aborted.</param>
+    /// <param name="fRetaining">Indicates whether retaining Commit was specified. Will be false.</param>
+    /// <param name="pNewUOW">Always null.</param>
+    /// <param name="hresult">Alawys S_OK.</param>
+    void Aborted(IntPtr pboidReason, bool fRetaining, Guid pNewUOW, uint hresult);
```

Comment:
COM interop defaults marshalling `bool` as `VARIANT_BOOL`. This should be marshalled as `WinBool` or `CBool`. Mark as `[MarshalAs(UnmanagedType.Bool)]`.